File: ruby/testcode/ssti_014.rb
require_relative 'shared'
require 'erb'

# vuln-code-snippet start ruby_ssti_erb_precompiled
def render_precompiled(req)
  name = escape_html(req.param('name'))
  template = ERB.new(File.read('templates/greeting.erb'))
  output = template.result_with_hash(name: name) # vuln-code-snippet safe-line ruby_ssti_erb_precompiled
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_ssti_erb_precompiled

File: php/testcode/ssti_014.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssti_plates_native
function ssti014(BenchmarkRequest $req): BenchmarkResponse {
    $templates = new League\Plates\Engine(__DIR__ . '/templates');
    $output = $templates->render('profile', ['name' => $req->param('name')]); // vuln-code-snippet safe-line php_ssti_plates_native
    return BenchmarkResponse::html($output);
}
// vuln-code-snippet end php_ssti_plates_native
